### PR TITLE
Add Docker connector

### LINKED ISSUE #238: Add Docker connector and converters

### PR DESCRIPTION
**Pull request by Roma-Kyrnis**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix #238

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/faros-destination/resources/spec.json [hunk 1]
```diff
@@ -303,6 +303,32 @@
...
+                "title": "Label Prefix",
+                "description": "Label prefix uses to construct build key and commit key for cicd_Artifact and cicd_ArtifactCommitAssociation from labels. Every label key start with this prefix. If prefix 'faros-' then label key is 'faros-build-id', 'faros-pipeline-id', etc."
+              },
+              "skip_tags": {
+                "type": "array",
+                "items": {
+                  "type": "string"
+                },
+                "title": "Skip tags",
+                "description": "The images with the listed tags will be ignored by the feed. Must be a comma separated list."
```
Comment (cjwooo) [inline destinations/faros-destination/resources/spec.json:324]: The wording is a bit confusing here. It's expecting an array but the description says comma-separated list?

Comment (cjwooo) [reply destinations/faros-destination/resources/spec.json:324]: In airbyte UI, you enter an array into the text field by pressing enter after each item.

File: destinations/faros-destination/src/converters/docker/tags.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+              uid: tag.name,
+              repository: repositoryKey,
+            },
+            commit: commitKey,
+          },
+        });
+      }
+    } else {
+      this.logger.info(
+        `kipped artifact in repo ${repositoryKey.uid} (uid: ${tag.name})`
```
Comment (cjwooo) [inline destinations/faros-destination/src/converters/docker/tags.ts:114]: Typo `kipped` => `skipped`.

File: sources/docker-source/test_files/spec.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        "description": "Fetch tags to exact repositories",
+        "item": {
+          "type": "string"
+        }
+      },
+      "authEndpoint": {
+        "type": "string",
+        "title": "Authorization endpoint",
+        "description": "Endpoint used to get access token",
+        "default": "tags/list"
```
Comment (cjwooo) [inline sources/docker-source/test_files/spec.json:37]: NIT: I think it'd look better if this were `/tags/list`.

**Review by cjwooo — CHANGES_REQUESTED** [review]

**Comment by cjwooo** [comment]
@Roma-Kyrnis Please resolve merge conflicts. (Can simply delete package-lock.json and recreate with `npm i`).

File: destinations/faros-destination/test/converters/docker.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    level: process.env.LOG_LEVEL ?? 'info',
+    prettyPrint: {levelFirst: true},
+  });
+  const mockttp = getLocal({debug: false, recordTraffic: false});
+  const catalogPath = 'test/resources/docker/catalog.json';
+  let configPath: string;
+  const streamNamePrefix = 'mytestsource__docker__';
+
+  beforeEach(async () => {
+    await mockttp.start({startPort: 30000, endPort: 50000});
```
Comment (cjwooo) [inline destinations/faros-destination/test/converters/docker.test.ts:23]: The logic for the faros destination has changed a bit with regards to config validation. Please change this line to https://github.com/faros-ai/airbyte-connectors/blob/main/destinations/faros-destination/test/converters/okta.test.ts#L23. I've updated the existing converter tests.

File: destinations/faros-destination/src/converters/docker/tags.ts (hunk 2 above)
Comment (Roma-Kyrnis) [reply destinations/faros-destination/src/converters/docker/tags.ts:114]: Thank you. I'll check next time text

File: destinations/faros-destination/test/converters/docker.test.ts (hunk 4 above)
Comment (Roma-Kyrnis) [reply destinations/faros-destination/test/converters/docker.test.ts:23]: Ok, I'll change

**Review by cjwooo — APPROVED** [review]

**Pull request merged**